### PR TITLE
Make firrtl-interpreter a submodule instead of depending on external snapshot

### PR DESCRIPTION
**Pull request by zhemao**
Recently, the sonatype URL we were using to pull in firrtl-interpreter stopped working. This commit adds firrtl-interpreter as a submodule and switches to including it as a project in the build.sbt instead.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -67,15 +67,23 @@ def isolateAllTests(tests: Seq[TestDefinition]) = tests map { test =>
   } toSeq
 
 // Subproject definitions begin
+//
```
Comment (abejgonzalez) [inline build.sbt]: We need to keep FIRRTL unmanaged. IIRC it is because RC also has it unmanaged.

Comment (zhemao) [reply build.sbt]: chisel-testers seems to break if we don't add firrtl as a dependency for it, though

Comment (zhemao) [reply build.sbt]: Oh nvm, I was reading that wrong. Seems like the problem is just that I didn't give the right dependencies to sha3.

Comment (zhemao) [reply build.sbt]: Nope, I was right the first time. RocketChip doesn't have a dependency on firrtl in its build.sbt, but firrtl-interpreter, chisel-testers, and treadle do, so we will have to manage a dependency for them it seems.

**Comment by seldridge** [comment]
Just a heads up, is there a reason why this doesn't use Treadle? 

FIRRTL Interpreter has some weird quirks that result in differing behavior vs. verilator/Treadle and is pushing towards getting deprecated. E.g., https://github.com/freechipsproject/chisel3/issues/993

**Comment by zhemao** [comment]
@seldridge We don't use firrtl-interpreter directly, but `chisel-testers` and some other submodules depend on it, so we have to include it here even if we aren't using it.

**Comment by seldridge** [comment]
@zhemao: Got it. It's just a build issue. As the interpreter gets deprecated, it'll get pulled out of here, too. Sounds great.

File: build.sbt [hunk 2]
```diff
@@ -67,15 +67,24 @@ def isolateAllTests(tests: Seq[TestDefinition]) = tests map { test =>
...
 
 // Subproject definitions begin
-
-// NB: FIRRTL dependency is unmanaged (and dropped in sim/lib)
+//
+// NB: FIRRTL should not be a managed dependency of chisel or rocketchip.
+// Instead, they will get firrtl from the JAR file placed in /lib
 lazy val chisel  = (project in rocketChipDir / "chisel3")
 
+lazy val firrtl = (project in file("tools/firrtl"))
```
Comment (davidbiancolin) [inline build.sbt]: this shouldn't be required.

Comment (davidbiancolin) [reply build.sbt]: The problem is probably that these tools are requesting a different version of firrtl than the one we are providing in lib

Comment (zhemao) [reply build.sbt]: But the FIRRTL jar we are providing in lib was generated from tools/firrtl, wasn't it?

Comment (grebe) [reply build.sbt]: What uses tools/firrtl? I would expect the firrtl in rocketchip to get used.

Comment (abejgonzalez) [reply build.sbt]: Everything should use the FIRRTL in `tools/firrtl`. We want to decouple RC's FIRRTL from the FIRRTL that we use so we could be ahead/behind what they provide.

File: build.sbt [hunk 3]
```diff
@@ -13,8 +13,8 @@ lazy val commonSettings = Seq(
   libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.5" % "test",
   libraryDependencies += "org.json4s" %% "json4s-jackson" % "3.6.1",
   libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value,
-  libraryDependencies += "edu.berkeley.cs" %% "firrtl-interpreter" % "1.2-SNAPSHOT",
   libraryDependencies += "com.github.scopt" %% "scopt" % "3.7.0",
+  libraryDependencies += "org.scala-lang.modules" % "scala-jline" % "2.12.1",
```
Comment (jwright6323) [inline build.sbt]: What is this?

File: project/plugins.sbt [hunk 4]
```diff
@@ -13,5 +13,6 @@ addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
 addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.6")
 addSbtPlugin("com.simplytyped" % "sbt-antlr4" % "0.8.1")
 addSbtPlugin("com.github.gseitz" % "sbt-protobuf" % "0.6.3")
+addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.4")
```
Comment (abejgonzalez) [inline project/plugins.sbt]: Similar to John's question... why is this needed?

File: build.sbt (hunk 3 above)
Comment (zhemao) [reply build.sbt]: It's a dependency that's needed by firrtl-interpreter.

https://github.com/freechipsproject/firrtl-interpreter/blob/master/build.sbt#L66

File: project/plugins.sbt (hunk 4 above)
Comment (zhemao) [reply project/plugins.sbt]: FIRRTL uses this plugin

https://github.com/freechipsproject/firrtl/blob/master/project/plugins.sbt#L23
